src/intelligence_layer/connectors/retrievers/hybrid_qdrant_in_memory_retriever.py

```diff
@@ -0,0 +1,206 @@
+from collections.abc import Sequence
+from concurrent.futures import ThreadPoolExecutor
+from typing import Optional
+
+from qdrant_client import QdrantClient
+from qdrant_client.http.models import Distance, PointStruct, VectorParams, models
+from qdrant_client.hybrid.fusion import reciprocal_rank_fusion
+
+from intelligence_layer.connectors.limited_concurrency_client import (
+    AlephAlphaClientProtocol,
+    LimitedConcurrencyClient,
+)
+from intelligence_layer.connectors.retrievers.base_retriever import (
+    Document,
+    DocumentChunk,
+    SearchResult,
+)
+from intelligence_layer.connectors.retrievers.qdrant_in_memory_retriever import (
+    QdrantInMemoryRetriever,
+    RetrieverType,
+)
+
+
+class HybridQdrantInMemoryRetriever(QdrantInMemoryRetriever):
+    """Search through documents stored in memory using hybrid (keyword + semantic) search.
+
+    This retriever uses a [Qdrant](https://github.com/qdrant/qdrant)-in-Memory vector store instance to store documents and their asymmetric embeddings.
+    When run, the given query is embedded using both a dense and sparse embedding model and scored against the documents in the collection to find the most relevant documents.
+    Finally, the retrievals are fused using the Reciprocal Rank Fusion algorithm.
+
+    Args:
+        documents: The sequence of documents to be made searchable.
+        k: The (top) number of documents to be returned by search.
+        client: Aleph Alpha client instance for running model related API calls.
+        threshold: The minimum value of the fusion rank score (combined cosine similarity and keyword similarity).
+        retriever_type: The type of retriever to be instantiated.
+            Should be `ASYMMETRIC` for most query-document retrieveal use cases, `SYMMETRIC` is optimized
+            for similar document retrieval.
+        distance_metric: The distance metric to be used for vector comparison.
+        sparse_model_name: The name of the sparse embedding model from `fastemebed` to be used.
+
+    Example:
+        >>> from intelligence_layer.connectors import LimitedConcurrencyClient, Document, HybridQdrantInMemoryRetriever
+        >>> client = LimitedConcurrencyClient.from_env()
+        >>> documents = [Document(text=t) for t in ["I do not like rain.", "Summer is warm.", "We are so back."]]
+        >>> retriever = HybridQdrantInMemoryRetriever(documents, 5, client=client)
+        >>> query = "Do you like summer?"
+        >>> documents = retriever.get_relevant_documents_with_scores(query)
+    """
+
+    MAX_WORKERS = 10
```

Comment:
There is no docstring about what MAX_WORKERS does, which is an attribute of the class. I think it would be a good idea to also move this into the init

Reply:
I c/p this from the `QdrantInMemoryRetriever` class, but can refactor as you suggest.